src/osd/PG.cc

```diff
@@ -3357,8 +3357,16 @@ bool PG::sched_scrub()
 
     //NOSCRUB so skip regular scrubs
     if ((osd->osd->get_osdmap()->test_flag(CEPH_OSDMAP_NOSCRUB) ||
-	 pool.info.has_flag(pg_pool_t::FLAG_NOSCRUB)) && !time_for_deep)
+	 pool.info.has_flag(pg_pool_t::FLAG_NOSCRUB)) && !time_for_deep) {
+      if (scrubber.reserved && !is_scrubbing()) {
```

Comment:
The !is_scrubbing() should always be true because it was already checked by the first test in the function:

  if (!(is_primary() && is_active() && is_clean() && !is_scrubbing())) {